remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/mappers/JsonRemoteMessageMapper.kt

```diff
@@ -0,0 +1,134 @@
+import com.duckduckgo.remote.messaging.impl.models.JsonRemoteMessage
+import timber.log.Timber
+
+class JsonRemoteMessageMapper {
+    private val smallMapper: (JsonContent) -> Content = { jsonContent ->
+        Small(
+            titleText = jsonContent.titleText.failIfEmpty(),
+            descriptionText = jsonContent.descriptionText.failIfEmpty()
+        )
+    }
+
+    private val mediumMapper: (JsonContent) -> Content = { jsonContent ->
+        Medium(
+            titleText = jsonContent.titleText.failIfEmpty(),
+            descriptionText = jsonContent.descriptionText.failIfEmpty(),
+            placeholder = jsonContent.placeholder.asPlaceholder()
+        )
+    }
+
+    private val bigMessageSingleAcionMapper: (JsonContent) -> Content = { jsonContent ->
+        BigSingleAction(
+            titleText = jsonContent.titleText.failIfEmpty(),
+            descriptionText = jsonContent.descriptionText.failIfEmpty(),
+            placeholder = jsonContent.placeholder.asPlaceholder(),
+            primaryActionText = jsonContent.primaryActionText.failIfEmpty(),
+            primaryAction = jsonContent.primaryAction!!.toAction()
+        )
+    }
+
+    private val bigMessageTwoAcionMapper: (JsonContent) -> Content = { jsonContent ->
+        BigTwoActions(
+            titleText = jsonContent.titleText.failIfEmpty(),
+            descriptionText = jsonContent.descriptionText.failIfEmpty(),
+            placeholder = jsonContent.placeholder.asPlaceholder(),
+            primaryActionText = jsonContent.primaryActionText.failIfEmpty(),
+            primaryAction = jsonContent.primaryAction!!.toAction(),
+            secondaryActionText = jsonContent.secondaryActionText.failIfEmpty(),
+            secondaryAction = jsonContent.secondaryAction!!.toAction()
+        )
+    }
+
+    private val messageMappers = mapOf(
+        Pair(SMALL.jsonValue, smallMapper),
+        Pair(MEDIUM.jsonValue, mediumMapper),
+        Pair(BIG_SINGLE_ACTION.jsonValue, bigMessageSingleAcionMapper),
+        Pair(BIG_TWO_ACTION.jsonValue, bigMessageTwoAcionMapper)
+    )
+
+    private val urlActionMapper: (JsonMessageAction) -> Action = {
+        Action.Url(it.value)
+    }
+
+    private val dismissActionMapper: (JsonMessageAction) -> Action = {
+        Action.Dismiss
+    }
+
+    private val playStoreActionMapper: (JsonMessageAction) -> Action = {
+        Action.PlayStore(it.value)
+    }
+
+    private val defaultBrowserActionMapper: (JsonMessageAction) -> Action = {
+        Action.DefaultBrowser
+    }
+
+    private val actionMappers = mapOf(
+        Pair(URL.jsonValue, urlActionMapper),
+        Pair(DISMISS.jsonValue, dismissActionMapper),
+        Pair(PLAYSTORE.jsonValue, playStoreActionMapper),
+        Pair(DEFAULT_BROWSER.jsonValue, defaultBrowserActionMapper)
+    )
+
+    fun map(jsonMessages: List<JsonRemoteMessage>): List<RemoteMessage> = jsonMessages.mapNotNull { it.map() }
```

Comment:
I would make this an extension fun of `List<JsonRemoteMessage>` and removing the necessity for the `JsonRemoteMessageMapper`.